scheduler/reconcile_util.go

```diff
@@ -510,21 +506,12 @@ func (a *allocNameIndex) NextCanaries(n uint, existing, destructive allocSet) []
 		}
 	}
 
-	// We have exhausted the preferred and free set, now just pick overlapping
+	// We have exhausted the preferred and free set. Pick starting from n to
+	// n+remainder, to avoid overlapping where possible.
```

Comment:
Can the comment also have an example, like if desired canaries = 3, and n = 2 it picks indexes 3, 4, 5.. I find this logic hard to follow. Also is there a unit test that asserts the names are what we expect after this specific change?

Reply:
Here is the unit test: https://github.com/hashicorp/nomad/pull/4206/files#diff-ecc3eb696252654b3d4c86abe54f849fR3152